Makefile

```diff
@@ -320,6 +322,30 @@ CONTROLLER_GEN = $(PROJECT_DIR)/bin/controller-gen
 controller-gen: ## Download controller-gen locally if necessary.
 	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install sigs.k8s.io/controller-tools/cmd/controller-gen
 
+GOOS = $(shell go env GOOS)
+GOARCH = $(shell go env GOARCH)
+
+ifeq ($(GOOS),darwin)
+    URL := https://github.com/koalaman/shellcheck/releases/download/v0.9.0/shellcheck-v0.9.0.darwin.x86_64.tar.xz
+else
+    ifeq ($(GOARCH),arm64)
+        URL := https://github.com/koalaman/shellcheck/releases/download/v0.9.0/shellcheck-v0.9.0.linux.aarch64.tar.xz
+    else
+        URL := https://github.com/koalaman/shellcheck/releases/download/v0.9.0/shellcheck-v0.9.0.linux.x86_64.tar.xz
+    endif
+endif
+
+SHELLCHECK_LINT = $(PROJECT_DIR)/bin/shellcheck
+.PHONY: shell-lint
+shell-lint:
+	@mkdir -p $(PROJECT_DIR)/bin && \
+	wget -q -O $(SHELLCHECK_LINT).tar.xz $(URL) && \
+	apt update && \
+	apt install xz-utils && \
```

Comment:
Please not to use apt since `apt` wouldn't work in some environments.
If we need to install some dependencies, using docker might be better like this: https://github.com/kubernetes/kubernetes/blob/0a31504ee114e363df9d69158c2e639858b8273b/hack/verify-shellcheck.sh#L117-L121

@Vandit1604 WDYT?